comp/netflow/server/integration_test.go

```diff
@@ -133,3 +141,137 @@ func TestNetFlow_IntegrationTest_SFlow5(t *testing.T) {
 
 	assertFlowEventsCount(t, port, srv, packetData, 7)
 }
+
+func TestNetFlow_IntegrationTest_AdditionalFields(t *testing.T) {
+	port, err := ndmtestutils.GetFreePort()
+	require.NoError(t, err)
+	var epForwarder forwarder.MockComponent
+	srv := fxutil.Test[Component](t, fx.Options(
+		testOptions,
+		fx.Populate(&epForwarder),
+		fx.Replace(
+			&nfconfig.NetflowConfig{
+				Enabled:                 true,
+				AggregatorFlushInterval: 1,
+				Listeners: []nfconfig.ListenerConfig{{
+					FlowType: common.TypeNetFlow9,
+					BindHost: "127.0.0.1",
+					Port:     port,
+					Mapping: []nfconfig.Mapping{
+						{
+							Field:       11,
+							Destination: "source.port", // Inverting source and destination port to test
+							Type:        common.Varint,
+						},
+						{
+							Field:       7,
+							Destination: "destination.port",
+							Type:        common.Varint,
+						},
+						{
+							Field:       32,
+							Destination: "icmp_type",
+							Type:        common.Bytes,
+						},
+					},
+				}},
+			},
+		),
+		setTimeNow,
+	)).(*Server)
+
+	flowData, err := testutil.GetNetFlow9Packet()
+	require.NoError(t, err, "error getting packet")
+
+	// Set expectations
+	testutil.ExpectPayloadWithAdditionalFields(t, epForwarder)
+	epForwarder.EXPECT().SendEventPlatformEventBlocking(gomock.Any(), "network-devices-metadata").Return(nil).Times(1)
+
+	assertFlowEventsCount(t, port, srv, flowData, 29)
+}
+
+func BenchmarkNetflowAdditionalFields(b *testing.B) {
```

Comment:
Benchmark results on my laptop :
```
BenchmarkNetflowAdditionalFields
BenchmarkNetflowAdditionalFields/goflow2_default
BenchmarkNetflowAdditionalFields/goflow2_default-10         	   30405	     39570 ns/op
BenchmarkNetflowAdditionalFields/goflow2_netflow_custom_state_without_custom_fields
BenchmarkNetflowAdditionalFields/goflow2_netflow_custom_state_without_custom_fields-10         	   29553	     40791 ns/op
BenchmarkNetflowAdditionalFields/goflow2_netflow_custom_state_with_custom_fields
BenchmarkNetflowAdditionalFields/goflow2_netflow_custom_state_with_custom_fields-10            	   21940	     54895 ns/op
```

Custom fields usage significantly slows the listener, but this can be solved by using more workers for listeners with a mapping config.